### PR TITLE
Documentation for foreman_templates 6.x

### PR DESCRIPTION
**Pull request by ares**
---

first commit is just a copy from 5.0, second commit describes changes, @xprazak2 mind to do a sanity check? Grammar check is welcome too, but we should get this out soon as 6.0.1 is already in nightlies, 6.0 is there for a while.

### CODE REVIEW

File: plugins/foreman_templates/6.0/index.md [hunk 1]
```diff
@@ -120,120 +120,98 @@ sudo -u foreman git config --global http.sslCAPath /path/to/ca/certs
...
-tree /tmp/my-repo/
-/tmp/my-repo/
-└── large-ptable.erb
-
-0 directories, 1 file
-
-# this will create a template named "[community] large-ptable"
-foreman-rake templates:import repo=/tmp/my-repo/ prefix='[community] '
-
+a ruby description of a JSON you get back follows
```
Comment (xprazak2) [inline plugins/foreman_templates/6.0/index.md]: Why use a Ruby syntax to describe JSON? Even the syntax for the snippet has json syntax highlighting. All it takes to turn it to a proper json is `s/=>/:/g` and `s/nil/null/g`.

File: plugins/foreman_templates/6.0/index.md [hunk 2]
```diff
@@ -120,120 +120,98 @@ sudo -u foreman git config --global http.sslCAPath /path/to/ca/certs
...
 ```sh
-foreman-rake templates:export
+curl -H "Accept:application/json,version=2" -H "Content-Type:application/json" -u user1:changeme -k https://foreman.example.com/api/v2/templates/export -X POST -d "{\"repo\":\"/another/repo\"}"
 ```
 
-If the `repo` is git repository, we clone it to a temp directory. Then we apply changes to template files and automatically create a commit with a generated message and push it back to origin. We do not delete files that are not present in Foreman database, so this should not be destructive. On the other hand if you want more control over the process or you want to use another SCM, local directory export is advised. To specify a local directory, just set the repo accordingly
- 
-```sh
-foreman-rake templates:export repo=/var/lib/my_templates
+and similar output for export endpoint
```
Comment (xprazak2) [inline plugins/foreman_templates/6.0/index.md]: Same as comment above.

File: plugins/foreman_templates/6.0/index.md [hunk 3]
```diff
@@ -120,120 +120,98 @@ sudo -u foreman git config --global http.sslCAPath /path/to/ca/certs
...
 
 ## 6.1 Known taxonomy problems when importing
 
-The import procedure uses template metadata to determine Organization/Location asssociations. When non-admin user initiates the import via API, it may be possible that metadata do not contain any Organization/Location the user is assigned to. In this case, import of the template fails with: `Invalid organizations selection, you must select at least one of yours`. To import such a template, update it's metadata so it contains at least one Organization/Location that user is assigned to.
+The import procedure uses template metadata to determine Organization/Location asssociations. When non-admin user initiates the import via API, it may be possible that metadata do not contain any Organization/Location the user is assigned to. In this case, import of the template fails with: `Invalid organizations selection, you must select at least one of yours`. To import such a template, update it's metadata so it contains at least one Organization/Location that user is assigned to. Or alternatively, specify organizations/locations as API parameter. Hammer plugin also exposes these arguments.
```
Comment (xprazak2) [inline plugins/foreman_templates/6.0/index.md]: Maybe we could have a simple example how to override taxonomies.

File: plugins/foreman_templates/6.0/index.md [hunk 4]
```diff
@@ -319,10 +297,105 @@ This is usually the case on systemd-enabled operating systems. Foreman service u
...
+
+The parsing process reads the content of the file. It expects the content of the template including some template metadata. The metadata are an ERB comment on top of the file, for illustration see some template in [community-templates repository](https://github.com/theforeman/community-templates/blob/develop/provisioning_templates/provision/kickstart_default.erb). It's in yaml format and specifies template name, kind, compatible operating systems, organizations, locations and possibly other. The template gets it's name based on information from metadata. Before the template is saved in database, `prefix` setting is applied to it's name. Then the `filter` regular expression is matched against the prefixed name. If the `filter` matches, templates is saved. If `negate` is set to `true`, the template would be saved only if `filter` regular expression didn't match the template name. If the template with a given name exists in database already, it's content is updated.
+
+The result of the whole import is printed on the STDOUT. The output contains list of created/updated templates as well as applied diffs.
```
Comment (xprazak2) [inline plugins/foreman_templates/6.0/index.md]: The progress of the whole import is logged.

File: plugins/foreman_templates/6.0/index.md [hunk 5]
```diff
@@ -319,10 +297,105 @@ This is usually the case on systemd-enabled operating systems. Foreman service u
...
+0 directories, 1 file
+
+# this will create a template named "[community] large-ptable"
+foreman-rake templates:import repo=/tmp/my-repo/ prefix='[community] '
+
+```
+
+# 8.2 Exporting
+
+If you want to edit templates directly in Foreman and then sync them back to external repositories, you need the exporting task. This plugin allows to export directly to git repository or local directory. For exporting to git repository, the user that's Foreman running under needs to have write (commit) access to this repository. If your settings is correctly configure, exporting can be triggered by running following command.
```
Comment (xprazak2) [inline plugins/foreman_templates/6.0/index.md]: `s/settings is correctly configure/settings are correctly configured/g`

**Comment by ares** [comment]
updated all

**Comment by xprazak2** [comment]
LGTM

**Comment by mmoll** [comment]
merged, díky @ares!

**Pull request merged**